test/runnable/testxmm.d

```diff
@@ -2390,8 +2390,32 @@ void test23009()
 
 /*****************************************/
 
+// Issue: https://issues.dlang.org/show_bug.cgi?id=23077
+
+float bug23077_1(float x) {
+    short i = *cast(short*)&x;
+    ++i;
+    return *cast(float*)&i;
+}
+
+float bug23077_2(float x) {
+    ushort i = *cast(ushort*)&x;
+    ++i;
+    return *cast(float*)&i;
+}
+
+float bug23077_3(float x) {
```

Comment:
Missing coverage from https://issues.dlang.org/show_bug.cgi?id=8060 which is related to this issue.